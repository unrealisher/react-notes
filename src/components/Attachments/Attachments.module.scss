.attachments {
  position: relative;
  padding: 12px 24px;
  padding-top: 40px;
  margin-top: -28px;
  background-color: rgba(216, 216, 216, 0.5);
  border-radius: 16px;
}

.list_link {
  display: flex;
  flex-direction: column;
}

.list_image {
  display: flex;
  justify-content: flex-end;
  position: relative;

  &::before {
    content: url(./images/attachment-image.svg);
    position: absolute;
    left: 0;
    top: calc(50% - 10px);
  }
}

.item_link {
  position: relative;
  padding: 10px 0;
  padding-left: 34px;

  &:first-child {
    padding-top: 0;
  }

  &::before {
    content: url(./images/attachment-link.svg);
    position: absolute;
    width: 24px;
    height: 24px;
    left: 0;
  }

  &:not(:last-child)::after {
    content: "";
    position: absolute;
    width: 100%;
    height: 1px;
    background-color: white;
    top: 100%;
    left: 0;
  }
}

.item_image:not(:last-child) {
  margin-right: 24px;
}

.image {
  width: 46px;
  height: 46px;
  border-radius: 4px;

  @media (min-width: 516px) {
    width: 64px;
    height: 64px;
  }
}

.link {
  margin-top: 2px;
  display: inline-block;
  color: black;
  font-weight: bold;

  @media (min-width: 516px) {
    margin: 0;
    display: inline;
  }
}
