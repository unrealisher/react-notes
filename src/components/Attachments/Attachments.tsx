import React from "react";

import IAttachment from "../../interfaces/IAttachment";

import styles from "./Attachments.module.scss";

interface IProps {
  attachments?: IAttachment[];
}

const Attachments = (props: IProps): JSX.Element => {
  const { attachments } = props;

  const getItems = (): JSX.Element => {
    if (attachments && attachments[0].type === "link") {
      return (
        <ul className={styles.list_link}>
          {attachments.map(item => {
            return (
              <li key={item.url} className={styles.item_link}>
                <a href={item.url} className={styles.link}>
                  {item.url && item.url.slice(8)}
                </a>
              </li>
            );
          })}
        </ul>
      );
    } else {
      return (
        <ul className={styles.list_image}>
          {attachments &&
            attachments.map(item => {
              return (
                <li key={item.url} className={styles.item_image}>
                  <img
                    src={item.url}
                    className={styles.image}
                    alt="Список изображений"
                  />
                </li>
              );
            })}
        </ul>
      );
    }
  };
  return <div className={styles.attachments}>{getItems()}</div>;
};

export default Attachments;
