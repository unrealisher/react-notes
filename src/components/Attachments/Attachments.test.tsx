import React from "react";
import { shallow } from "enzyme";
import { Attachments } from "./Attachments";

describe("Attachments", () => {
  const attach = [
    { type: "link", url: "https://yandex.ru" },
    { type: "link", url: "https://ya.ru" }
  ];
  it("correct render", () => {
    const component = shallow(<Attachments attachments={attach} />);
    expect(component).toMatchSnapshot();
  });
});
