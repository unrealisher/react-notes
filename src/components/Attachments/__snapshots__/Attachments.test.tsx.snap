// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Attachments
    attachments={
      Array [
        Object {
          "type": "link",
          "url": "https://yandex.ru",
        },
        Object {
          "type": "link",
          "url": "https://ya.ru",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list_link"
      >
        <li
          className="item_link"
        >
          <a
            className="link"
            href="https://yandex.ru"
          >
            yandex.ru
          </a>
        </li>
        <li
          className="item_link"
        >
          <a
            className="link"
            href="https://ya.ru"
          >
            ya.ru
          </a>
        </li>
      </ul>,
      "className": "attachments",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item_link"
          >
            <a
              className="link"
              href="https://yandex.ru"
            >
              yandex.ru
            </a>
          </li>,
          <li
            className="item_link"
          >
            <a
              className="link"
              href="https://ya.ru"
            >
              ya.ru
            </a>
          </li>,
        ],
        "className": "list_link",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "https://yandex.ru",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="link"
              href="https://yandex.ru"
            >
              yandex.ru
            </a>,
            "className": "item_link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "yandex.ru",
              "className": "link",
              "href": "https://yandex.ru",
            },
            "ref": null,
            "rendered": "yandex.ru",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "https://ya.ru",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="link"
              href="https://ya.ru"
            >
              ya.ru
            </a>,
            "className": "item_link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ya.ru",
              "className": "link",
              "href": "https://ya.ru",
            },
            "ref": null,
            "rendered": "ya.ru",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list_link"
        >
          <li
            className="item_link"
          >
            <a
              className="link"
              href="https://yandex.ru"
            >
              yandex.ru
            </a>
          </li>
          <li
            className="item_link"
          >
            <a
              className="link"
              href="https://ya.ru"
            >
              ya.ru
            </a>
          </li>
        </ul>,
        "className": "attachments",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="item_link"
            >
              <a
                className="link"
                href="https://yandex.ru"
              >
                yandex.ru
              </a>
            </li>,
            <li
              className="item_link"
            >
              <a
                className="link"
                href="https://ya.ru"
              >
                ya.ru
              </a>
            </li>,
          ],
          "className": "list_link",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "https://yandex.ru",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="link"
                href="https://yandex.ru"
              >
                yandex.ru
              </a>,
              "className": "item_link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "yandex.ru",
                "className": "link",
                "href": "https://yandex.ru",
              },
              "ref": null,
              "rendered": "yandex.ru",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "https://ya.ru",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="link"
                href="https://ya.ru"
              >
                ya.ru
              </a>,
              "className": "item_link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ya.ru",
                "className": "link",
                "href": "https://ya.ru",
              },
              "ref": null,
              "rendered": "ya.ru",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
